dev/tools/vitool/lib/vitool.dart

```diff
@@ -0,0 +1,551 @@
+    if (runtimeType != other.runtimeType)
+      return false;
+    final SvgPathCommand typedOther = other;
+    return type == typedOther.type
+        && const ListEquality<Point<double>>().equals(points, typedOther.points);
+  }
+
+  @override
+  int get hashCode => type.hashCode ^ points.hashCode;
+
+  @override
+  String toString() {
+    return 'SvgPathCommand(type: $type, points: $points)';
+  }
+}
+
+class SvgPathCommandBuilder {
+  static const Map<String, Null> kRelativeCommands = const <String, Null> {
+    'c': null,
+    'l': null,
+    'm': null,
+    't': null,
+    's': null
+  };
+
+  Point<double> lastPoint = const Point<double>(0.0, 0.0);
+  Point<double> subPathStartPoint = const Point<double>(0.0, 0.0);
+
+  SvgPathCommand build(String type, List<Point<double>> points) {
+    List<Point<double>> absPoints = points;
+    if (_isRelativeCommand(type)) {
+      absPoints = points.map((Point<double> p) => p + lastPoint).toList();
+    }
+
+    if (type == 'M' || type == 'm')
+      subPathStartPoint = absPoints.last;
+
+    if (type == 'Z' || type == 'z')
+      lastPoint = subPathStartPoint;
+    else
+      lastPoint = absPoints.last;
+
+    return new SvgPathCommand(type.toUpperCase(), absPoints);
+  }
+
+  static bool _isRelativeCommand(String type) {
+    return kRelativeCommands.containsKey(type);
+  }
+}
+
+List<double> _pointsToVector3Array(List<Point<double>> points) {
+  final List<double> result = new List<double>(points.length * 3);
+  for (int i = 0; i < points.length; i += 1) {
+    result[i * 3] = points[i].x;
+    result[i * 3 + 1] = points[i].y;
+    result[i * 3 + 2] = 1.0;
+  }
+  return result;
+}
+
+List<Point<double>> _vector3ArrayToPoints(List<double> vector) {
+  final int numPoints = (vector.length / 3).floor();
+  final List<Point<double>> points = new List<Point<double>>(numPoints);
+  for (int i = 0; i < numPoints; i += 1) {
+    points[i] = new Point<double>(vector[i*3], vector[i*3 + 1]);
+  }
+  return points;
+}
+
+/// Represents a transformation to apply on an SVG subtree.
+///
+/// This includes more transforms than the ones described by the SVG transform
+/// attribute, e.g opacity.
+class _Transform {
+
+  /// Constructs a new _Transform, default arguments create a no-op transform.
+  _Transform({Matrix3 transformMatrix, this.opacity = 1.0}) :
+      this.transformMatrix = transformMatrix ?? new Matrix3.identity();
+
+  final Matrix3 transformMatrix;
+  final double opacity;
+
+  _Transform applyTransform(_Transform transform) {
+    return new _Transform(
+        transformMatrix: transform.transformMatrix.multiplied(transformMatrix),
+        opacity: transform.opacity * opacity,
+    );
+  }
+}
+
+final RegExp _transformCommand = new RegExp(' *([^)]+)\\(([^)]*)\\)');
+
+Matrix3 _parseSvgTransform(String transform){
+  final Iterable<Match> matches =_transformCommand.allMatches(transform).toList().reversed;
+  Matrix3 result = new Matrix3.identity();
+  for (Match m in matches) {
```

Comment:
This won't properly catch bogus transforms. Like, if the value was `transform="foo"` it wouldn't match anything. If it was `transform="foo() bar baz()"` it would see the "foo" and "baz" parts but not "bar". If it was `transform="x() !!!"`it wouldn't notice the `!!!`.

Reply:
Added stricter validation.